app/services/concerns/xccdf/util.rb

```diff
@@ -30,11 +30,22 @@ def save_all_benchmark_info
 
       def save_all_test_result_info
         save_host
-        @host_profile = save_profile_host
+        save_profile_host
         save_rule_results
+        associate_rules_from_rule_results
         invalidate_cache
       end
 
+      def set_openscap_parser_data
+        @op_benchmark = @test_result_file.benchmark
+        @op_test_result = @test_result_file.test_result
+        @op_profiles = @op_benchmark.profiles
+        @op_rules = @op_benchmark.rules
+        @op_rule_references =
+          @op_benchmark.rule_references.reject { |rr| rr.label.empty? }
+        @op_rule_results = @op_test_result.rule_results
+      end
```

Comment:
I decided to move this to util because it's required to set up all the rest of the xccdf parsing methods (and we can conveniently use it in tests).